src/history/hs.c

```diff
@@ -373,7 +376,7 @@ __hs_insert_updates_verbose(WT_SESSION_IMPL *session, WT_BTREE *btree)
 static int
 __hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, const uint32_t btree_id,
   const WT_ITEM *key, const WT_UPDATE *upd, const uint8_t type, const WT_ITEM *hs_value,
-  WT_TIME_PAIR stop_ts_pair)
+  WT_TIME_PAIR stop_ts_pair, bool *deleted_prev_hs)
```

Comment:
Do we really need this flag? In an update chain even if we have two updates with zero timestamp, still we get here only one update. So I don't see the use of it. Am I missing anything?

Reply:
6327744

I've removed the parameter but for a different reason. Why will there be only one update with a zero timestamp here? They can come from different transactions.

The reason I removed it is because if we have an update list with timestamps 0-1-0-1, we won't delete the history store content for the original 1 which is incorrect.